### ISSUE TITLE
incompatible hsdev detected

### ISSUE THREAD

**Issue by pascaldanek** [SublimeHaskell/SublimeHaskell#411]
Hi, 

first test of SublimeHaskell trying to setup an haskell dev env on Mac. 

On Sublime startup I get:

```
Backend configuration: "Remote hsdev@localhost:4567"
Incompatible hsdev, detected version 0.3.1.2
Version should be ≥ 0.2.0.0 and < 0.3.0.0
```

Sublime version: 3.0 build 3143
SublimeHaskell version: 2.1.25

Is it the expected behavior ? Do I have to setup a specific version of hsdev ?

Thanks,

**Comment by bscottm** [comment]
@pascaldanek: Please scroll down in the release notes. There's a note for that. :-)

**Comment by pascaldanek** [comment]
Very sorry for the noise guys.

**Comment by Himbeereis** [comment]
Same problem here. Where do I find the release notes? thanks

**Comment by andrewdoh** [comment]
@Himbeereis its in the package control messages that pops up when you open sublime.

## In The Meantime... `hsdev 0.3.0.1`: Do Not Use
  
  Recently, `hsdev 0.3.0.1` was released, . If you installed `hsdev` since New Year's 2018, either via `cabal` or `stack`
  
      cabal install hsdev
  
  or
  
      stack install hsdev
  
  then type the following at your shell or command window's prompt:
  
      hsdev version
  
  If the version number does not start with `0.2`, you inadvertently installed an incompatible version. Version `0.3.0.1` support will be released soon, pending testing.
  
  To install the `0.2` series version that is compatible with _SublimeHaskell_:
  
      cabal install hsdev-0.2.5.1
  
  If you try:
  
      stack install hsdev-0.2.5.1
  
  `stack` will complain about `haddock-api` and `hdocs` incompatibilities (the `stack.yaml` file probably needs updating.) Stick with `cabal install` for the time being.

**Comment by Searge** [comment]
For a week I'm trying to onfigure that, but that dependencies is killing me :-)

```bash
❯ stack install hsdev-0.2.5.1

Error: While constructing the build plan, the following exceptions were encountered:

In the dependencies for hsdev-0.2.5.1:
    haddock-api must match >=2.18.0 && <2.19.0, but the stack configuration has no specified version  (latest matching
                version is 2.18.1)
    haskell-src-exts-1.20.2 from stack configuration does not match >=1.18.0 && <1.20.0  (latest matching version
                            is 1.19.1)
    hdocs must match >=0.5.0, but the stack configuration has no specified version  (latest matching version is 0.5.3.0)
    hlint-2.1.5 from stack configuration does not match >=1.9.13 && <2.1  (latest matching version is 2.0.15)
needed since hsdev is a build target.

~
❯ mate ~/.stack/global-project/stack.yaml

~
❯ stack install hsdev-0.2.5.1

Error: While constructing the build plan, the following exceptions were encountered:

In the dependencies for haddock-api-2.18.1:
    haddock-library-1.4.5 from stack configuration does not match ==1.4.4.*  (latest matching version is 1.4.4)
needed due to hsdev-0.2.5.1 -> haddock-api-2.18.1

In the dependencies for haskell-src-exts-util-0.2.3:
    haskell-src-exts-1.19.1 from stack configuration does not match >=1.20.0  (latest matching version is 1.20.2)
needed due to hsdev-0.2.5.1 -> haskell-src-exts-util-0.2.3

In the dependencies for hlint-2.0.15:
    extra-1.6.6 from stack configuration does not match >=1.4.9 && <=1.6.5  (latest matching version is 1.6.5)
    haskell-src-exts-1.19.1 from stack configuration does not match >=1.20 && <1.21  (latest matching version is 1.20.2)
needed due to hsdev-0.2.5.1 -> hlint-2.0.15

Some different approaches to resolving this:

  * Set 'allow-newer: true' to ignore all version constraints and build anyway.

  * Consider trying 'stack solver', which uses the cabal-install solver to attempt to find some working build
    configuration. This can be convenient when dealing with many complicated constraint errors, but results may be
    unpredictable.

  * Recommended action: try adding the following to your extra-deps in /Users/searge/.stack/global-project/stack.yaml:

- extra-1.6.5
- haddock-library-1.4.4
- haskell-src-exts-1.20.2

Plan construction failed.

~
❯ stack install hsdev-0.2.5.1

Error: While constructing the build plan, the following exceptions were encountered:

In the dependencies for haskell-src-exts-util-0.2.3:
    haskell-src-exts-1.19.1 from stack configuration does not match >=1.20.0  (latest matching version is 1.20.2)
needed due to hsdev-0.2.5.1 -> haskell-src-exts-util-0.2.3

In the dependencies for hlint-2.0.15:
    haskell-src-exts-1.19.1 from stack configuration does not match >=1.20 && <1.21  (latest matching version is 1.20.2)
needed due to hsdev-0.2.5.1 -> hlint-2.0.15

Some different approaches to resolving this:

  * Set 'allow-newer: true' to ignore all version constraints and build anyway.

  * Consider trying 'stack solver', which uses the cabal-install solver to attempt to find some working build
    configuration. This can be convenient when dealing with many complicated constraint errors, but results may be
    unpredictable.

  * Recommended action: try adding the following to your extra-deps in /Users/searge/.stack/global-project/stack.yaml:

- haskell-src-exts-1.20.2

Plan construction failed.
```

As you see, it's asking for `- haskell-src-exts-1.19.1`, and when you give it, it asking for `- haskell-src-exts-1.20.2` :D

**Comment by ratijas** [comment]
### HOW TO install the correct version of hsdev in one command

`$ stack --resolver lts-9.1 --install-ghc install hsdev-0.2.5.1`

Guaranteed to work, since resolver `lts-9.1` was released the same day as `hsdev-0.2.5.1`.
